draft-ietf-mls-protocol.md

```diff
@@ -2186,13 +2157,18 @@ struct {
 } ReInit;
 ~~~~~
 
-A member of the group applies a ReInit message by waiting for the committer to
+A member of the group applies a ReInit proposal by waiting for the committer to
 send the Welcome message and by checking that the `group_id` and the parameters
 of the new group corresponds to the ones specified in the proposal. The Welcome
-message MUST specify a `PreSharedKey` with `psktype = reinit` and with
+message MUST specify exactly one pre-shared key with `psktype = reinit`, and with
 `psk_group_id` and `psk_epoch` equal to the `group_id` and `epoch` of the
 existing group after the Commit containing the `reinit` Proposal was processed.
+The Welcome message may specify the inclusion of other pre-shared keys with a
+`psktype` different from `reinit`.
 
+If a ReInit proposal is included in a Commit, it MUST be the only proposal
+referenced by the Commit. The `version` field in the ReInit proposal MUST be no
```

Comment:
This sounds like existing proposals should be dropped, which should not be the case. I think it better to mandate an extra Commit that is issued when there are no more pending proposals.

Reply:
Yes, the ReInit must be dropped or the rest of the proposals must be dropped. Applications can do what you say automatically: Commit outstanding proposals -> send new ReInit -> Commit ReInit. There's no reason to mandate it

Reply:
Why does it have to be the only proposal in the commit? I mean, updates are going to be useless, but Adds and Removes can still be processed.

Reply:
Discussed at virtual interim 2020-10-20:
* ReInit MUST be only proposal
* If non-ReInit and ReInit received in same epoch, SHOULD prefer non-ReInit
* Options for recovering from contention: committer re-originates or original proposer retries